addons/sourcemod/scripting/include/shop/players.inc

```diff
@@ -426,13 +426,13 @@ native bool Shop_RemoveClientItem(int client, ItemId item_id, int count = 0);
  *
  *	@param client				Client index.
  *	@param item_id				The item id.
- *	@param value				Count if the item is finite and duration if the item is togglable or non-togglable.
+ *	@param value				Count if the item is finite and duration if the item is togglable or non-togglable, -1 eternal
```

Comment:
What does it means, `-1 external`? How it affect on logic? If **Item_Togglable** < 1 makes sense to make item duration infinite. If **Item_Finite** or **Item_BuyOnly,** parameter can't be < 1, it will automatically be set to 1, as minimal count is 1.

Reply:
"-1 eternal" means - infinite if set to "-1". 
As I understand, if this value is less than 1, the duration is infinite. 
From a logic point of view, it is useless to call the native Shop_GiveClientItem to set duration or quantity to 0. (Of course, you know, some people could try, but we rely on common sense). So it is possible to use "0" as a flag to use a value from the item's defaults (as it is used in the current version) or, if it is set to another value except 0, so to use the value received from the native. And "-1" value for the duration is the hint for the user who wants the infinite duration. If he tries to set the negative value for non-togglable, it is against the comment sense and "1" will be set automatically as you mentioned.

Reply:
All other logic does not affect, because other functions do not call "GiveItem() with the third parameter now. If it may be called in future, we can add the description to use "-1" for infinite duration. If the "0" in the item defaults, it does not affect too.